cancel_token/token.py

```diff
@@ -132,21 +143,35 @@ def cancel_not_done(fut: 'asyncio.Future[None]') -> None:
                 return_when=asyncio.FIRST_COMPLETED,
                 loop=self.loop,
             )
-        except asyncio.futures.CancelledError:
-            # Since we use return_when=asyncio.FIRST_COMPLETED above, we can be sure none of our
-            # futures will be done here, so we don't need to check if any is done before cancelling.
+        except asyncio.CancelledError as err:
+            # Since we use return_when=asyncio.FIRST_COMPLETED above, we can
+            # be sure none of our futures will be done here, so we don't need
+            # to check if any is done before cancelling.
             for future in futures:
                 future.cancel()
-            raise
-        for task in pending:
-            task.cancel()
+            for future in futures:
+                try:
+                    await future
+                except asyncio.CancelledError:
+                    pass
+            raise err
+        else:
+            for future in pending:
+                future.cancel()
+            for future in pending:
+                try:
+                    await future
+                except asyncio.CancelledError:
+                    pass
+
         if not done:
-            raise TimeoutError()
+            raise asyncio.TimeoutError()
+
         if self.triggered_token is not None:
             # We've been asked to cancel so we don't care about our future, but we must
             # consume its exception or else asyncio will emit warnings.
             for task in done:
-                task.exception()
+                task.result()
```

Comment:
2. We want the exception actually raised here? Seems like it needs a test and/or explanation.

Reply:
This is covered in `test_finished_task_with_exceptions_is_raised_on_cancellation`